draft-bbs-signatures.md

```diff
@@ -729,39 +735,60 @@ Procedure:
+This operation describes how to hash an arbitrary octet string to `n` scalar values in the multiplicative group of integers mod r (i.e., values in the range [1, r-1]).  This procedure acts as a helper function, used internally in various places within the operations described in the spec. To map a message to a scalar that would be passed as input to the [Sign](#sign), [Verify](#verify), [ProofGen](#proofgen) and [ProofVerify](#proofgen) functions, one must use [MapMessageToScalarAsHash](#mapmessagetoscalar) instead.
 
-The `hash_to_scalar` procedure hashes elements using an extendable-output function (xof). Applications not wishing to use an xof may use `hash_to_field` defined in Section 5.2 of [@!I-D.irtf-cfrg-hash-to-curve], combined with `expand_message_xmd` defined in Section 5.3.1 of the same document, in place of `hash_to_scalar`. In that case, every element outputted by `hash_to_field` that is equal to 0 MUST be rejected. If that occurs, one should calculate more field elements (using `hash_to_field`), until they get `n` non-zero elements (for example, if there is only one 0 in the output of `hash_to_field(msg, 2)` one must try to calculate `hash_to_field(msg, 3)` etc.).
+This operation makes use of expand\_message defined in [@!I-D.irtf-cfrg-hash-to-curve], in a similar way used by the hash\_to\_field operation of Section 5 from the same document (with the additional checks for getting a scalar that is 0). Note that, if an implementer wants to use hash\_to\_field here instead, they MUST use the multiplicative group of integers mod r (Fr), as the target group (F). However, the hash\_to\_curve ciphersuites used by this document, makes use of hash\_to\_field with the target group being the multiplicative group of integers mod p (Fp). For completeness, we define here the operation making use of the expand\_message function, that will be defined by the hash-to-curve ciphersuite used. If someone also has a hash\_to\_field implementation available, with the target group been Fr, they can use this instead (adding the check for a scalar been 0).
 
 ```
-result = hash_to_scalar(msg_octets, n)
+result = hash_to_scalar(msg_octets, count)
 
 Inputs:
 
 - msg_octets (REQUIRED), octet string. The message to be hashed.
-- n (REQUIRED), non-negative integer. The number of scalars to output.
+- count (REQUIRED), an integer greater or equal to 1. The number of
+                    scalars to output.
 
 Parameters:
 
-- r (REQUIRED), non-negative integer. The prime order of the G_1 and G_2 groups,
-     defined by the ciphersuite.
+- expand_message, as defined by the hash-to-curve suite uniquely
+                  identified by the hash_to_curve_id ciphersuite
+                  parameter.
+- dst, octet string. the separation tag defined by the
```

Comment:
I'm not sure what DST value this is?

Reply:
Ah, looks like it's the hash-to-curve ciphersuite identifier. There's a small issue that `hash_to_curve` is defined by the ciphersuite but `hash_to_curve_id` is referenced here and in `create_generators`

Reply:
Updated! Thank you! (named the ciphersuite parameter `hash_to_curve_suite` and used it in `create_generators` and `hash_to_scalar`)